src/main/java/org/jabref/gui/search/GlobalSearchBar.java

```diff
@@ -235,6 +242,17 @@ private void initSearchModifierButtons() {
         fulltextButton.setTooltip(new Tooltip(Localization.lang("Fulltext search")));
         initSearchModifierButton(fulltextButton);
         fulltextButton.selectedProperty().addListener((obs, oldVal, newVal) -> {
+            if (!filePreferences.shouldFulltextIndexLinkedFiles() && newVal) {
+                boolean enableFulltextSearch = dialogService.showConfirmationDialogAndWait(Localization.lang("Fulltext search"), Localization.lang("Fulltext search requires the setting 'Automatically index all linked files for fulltext search' to be enabled. Do you want to enable indexing now?"), Localization.lang("Enable indexing"), Localization.lang("Keep disabled"));
+
+                LibraryTab libraryTab = tabContainer.getCurrentLibraryTab();
+                if (libraryTab != null && enableFulltextSearch) {
```

Comment:
We should try to disable the button if no library is active. Refs https://github.com/JabRef/jabref/issues/11837.

Follow-up I would say.

Reply:
If we disable it, the new dialog would never come up

Reply:
Search is disabled if libray is not active 
<img width="795" alt="grafik" src="https://github.com/user-attachments/assets/1791bc11-21a8-42ae-a743-e6fa83ffb5f7">

Reply:
Then why this null check? 😅

Reply:
Intellij suggested that because library tab is nullabel...

Reply:
> If we disable it, the new dialog would never come up

The button should be disabled when preferences are off. The dialog will appear if the preferences are disabled and clicks to enable the button. If accepted to enable indexing, the button and preferences should both be enabled; otherwise, both should remain disabled.

Reply:
> > If we disable it, the new dialog would never come up
> 
> The button should be disabled when preferences are off. The dialog will appear if the preferences are disabled and clicks to enable the button. If accepted to enable indexing, the button and preferences should both be enabled; otherwise, both should remain disabled.

I understand  "clicks to enable the button" as a "click" int he preference.

The thing Christoph implemented was the intention of the whole thing. If we did as you described, we would not implement https://github.com/JabRef/jabref/issues/9491#issuecomment-1369188282.

Reply:
> Intellij suggested that because library tab is nullable.. (reduce warnings in the code=.

CC @calixtus. Annotation introduced at #10578. Follow-up: Document when this thing can be null. We should try not to pass `null`.

https://github.com/JabRef/jabref/blob/e395853e192951d5d90e7254fab68c691787e598/src/main/java/org/jabref/gui/LibraryTabContainer.java#L15

Reply:
> I understand "clicks to enable the button" as a "click" int he preference.

The toggle button in the search bar.

Reply:
please test the latest changes should be okay now